addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/AirVisualNodeBindingConstants.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.airvisualnode;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.DefaultSystemChannelTypeProvider;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+
+/**
+ * The {@link AirVisualNodeBindingConstants} class defines common constants, which are
+ * used across the whole binding.
+ *
+ * @author Victor Antonovich - Initial contribution
+ */
+@NonNullByDefault
+public class AirVisualNodeBindingConstants {
+
+    public static final String BINDING_ID = "airvisualnode";
+
+    // List of all Thing Type UIDs
+    public static final ThingTypeUID THING_TYPE_AVNODE = new ThingTypeUID(BINDING_ID, "avnode");
+
+    // List of all Channel ids
+    public static final String CHANNEL_CO2 = "co2";
+    public static final String CHANNEL_HUMIDITY = "humidity";
+    public static final String CHANNEL_AQI_US = "aqi";
+    public static final String CHANNEL_PM_25 = "pm_25";
+    public static final String CHANNEL_TEMP_CELSIUS = "temperature";
+    public static final String CHANNEL_TIMESTAMP = "timestamp";
+    public static final String CHANNEL_USED_MEMORY = "used_memory";
+    public static final String CHANNEL_BATTERY_LEVEL = DefaultSystemChannelTypeProvider.SYSTEM_CHANNEL_BATTERY_LEVEL
+            .getUID().getId();
+    public static final String CHANNEL_WIFI_STRENGTH = DefaultSystemChannelTypeProvider.SYSTEM_CHANNEL_SIGNAL_STRENGTH
+            .getUID().getId();
+
+    // List of all supported Thing UIDs
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
```

Comment:
Afair, `Set`'s should be annotated to have a nullable generic, because otherwise the compiler might assume that a set cannot return null but it can.

@maggu2810 or am I mistaken now?

Reply:
@martinvw You are right, but where is that `@Nullable` annotation that you are referring to?

Reply:
Missing and the class is annotated to be not Null

Reply:
Ah now I understand my misunderstanding of your comment...

`Set`s should **not** be annotated to have a `@Nullable` generic, because that would express that we allow to add `null` to the collection.

The problem you are mentioning is that a `get(item)` on a `Set` might always return `null` if the requested `item` is not in the `Set`. But that should be covered by external annotations which annotate (in this case) the JDT. But we do not have them, both PRs https://github.com/eclipse/smarthome/pull/4513 and https://github.com/eclipse/smarthome/pull/4217 were closed.

Long story short: The line here is correct.

Reply:
Nothing to add to @triller-telekom's explanation. :wink:

Reply:
Great thanks guys!